Lib/_colorize.py

```diff
@@ -0,0 +1,64 @@
+import io
+import os
+import sys
+
+COLORIZE = True
+
+
+class ANSIColors:
+    BOLD_GREEN = "\x1b[1;32m"
+    BOLD_MAGENTA = "\x1b[1;35m"
+    BOLD_RED = "\x1b[1;31m"
+    GREEN = "\x1b[32m"
+    GREY = "\x1b[90m"
+    MAGENTA = "\x1b[35m"
+    RED = "\x1b[31m"
+    RESET = "\x1b[0m"
+    YELLOW = "\x1b[33m"
+
+
+NoColors = ANSIColors()
+
+for attr in dir(NoColors):
+    if not attr.startswith("__"):
+        setattr(NoColors, attr, "")
+
+
+def get_colors(colorize: bool = False) -> ANSIColors:
+    if colorize or can_colorize():
+        return ANSIColors()
+    else:
```

Comment:
nit: `else` is unnecessary because we return on condition above. Normally it's even getting caught by the linter ( https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/no-else-return.html )

Reply:
It can be argued that's it's clearer with the `else`:

https://testing.googleblog.com/2023/09/else-nuances.html

Reply:
> It can be argued that's it's clearer with the `else`:
> 
> https://testing.googleblog.com/2023/09/else-nuances.html

That was an interesting read, thanks :) I guess it's resolved then since returning colorizing class is core responsibility of this function.